src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProjectProvider.cs

```diff
@@ -0,0 +1,171 @@
+                async () =>
+                {
+                    await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    return vsServiceProvider.GetService<SComponentModel, IComponentModel>();
+                },
+                _threadingService.JoinableTaskFactory);
+        }
+
+        public bool TryCreateNuGetProject(
+            IVsProjectAdapter vsProjectAdapter,
+            ProjectProviderContext context,
+            bool forceProjectType,
+            out NuGetProject result)
+        {
+            Assumes.Present(vsProjectAdapter);
+            Assumes.Present(context);
+
+            _threadingService.ThrowIfNotOnUIThread();
+
+            result = null;
+
+            var projectServices = _threadingService.ExecuteSynchronously(
+                () => TryCreateProjectServicesAsync(
+                    vsProjectAdapter, forceCreate: forceProjectType));
+
+            if (projectServices == null)
+            {
+                return false;
+            }
+
+            result = new LegacyPackageReferenceProject(
+                vsProjectAdapter,
+                vsProjectAdapter.ProjectId,
+                projectServices,
+                _threadingService);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Is this project a non-CPS package reference based csproj?
+        /// </summary>
+        private async Task<INuGetProjectServices> TryCreateProjectServicesAsync(
+            IVsProjectAdapter vsProjectAdapter, bool forceCreate)
+        {
+            var componentModel = await _componentModel.GetValueAsync();
+
+            // Check for RestoreProjectStyle property
+            var restoreProjectStyle = await vsProjectAdapter.BuildProperties.GetPropertyValueAsync(
+                ProjectBuildProperties.RestoreProjectStyle);
+
+            if (vsProjectAdapter.IsDeferred)
+            {
+                if (!forceCreate && 
+                    !PackageReference.Equals(restoreProjectStyle, StringComparison.OrdinalIgnoreCase))
+                {
+                    var packagesConfigFilePath = Path.Combine(vsProjectAdapter.FullPath, NuGetConstants.PackageReferenceFile);
+                    if (await vsProjectAdapter.EntityExistsAsync(packagesConfigFilePath))
+                    {
+                        return null;
+                    }
+
+                    var projectJsonFilePath = Path.Combine(vsProjectAdapter.FullPath, NuGetConstants.PackageSpecFileName);
+                    if (await vsProjectAdapter.EntityExistsAsync(projectJsonFilePath))
+                    {
+                        return null;
+                    }
+                }
+
+                var projectServices = componentModel.GetService<VsProjectSystemServices>();
+                Assumes.Present(projectServices);
+
+                return new DeferredProjectServicesProxy(
+                    vsProjectAdapter,
```

Comment:
When a project gets loaded for real we will create new instance of `NuGetProject` in response to the project-added event.